### PR TITLE
feat(dlm): adapts to the task framework and supports OceanBase Oracle mode

### PR DESCRIPTION
**Pull request by guowl3**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
To enhance the stability of task execution, data cleaning and data archiving tasks are integrated into the task framework for execution.At the same time, data archiving and data cleaning have added support for the OceanBase Oracle data source to meet customer needs.
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
It will be merged after SDK release.

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/CloudDLMJobStore.java [hunk 1]
```diff
@@ -55,26 +135,89 @@ public void bindGeneratorToJob(String s, TaskGenerator taskGenerator) throws SQL
...
 
     @Override
     public void storeJobStatistic(JobMeta jobMeta) throws JobSqlException {
-
+        jobMeta.getJobStat().buildReportData();
     }
 
     @Override
     public List<TaskMeta> getTaskMeta(JobMeta jobMeta) throws SQLException {
+        if (enableBreakpointRecovery) {
```
Comment (krihy) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/CloudDLMJobStore.java:148]: may bee add a  BreakpointRecoveryJobStore for handle enableBreakpointRecovery

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/DefaultJobTerminateListener.java [hunk 2]
```diff
@@ -61,35 +61,35 @@ public class DefaultJobTerminateListener extends AbstractEventListener<JobTermin
...
                     event.getStatus().convertTaskStatus());
             if (row >= 1) {
-                log.info("Update scheduleTask successfully, scheduleTaskId={}, status={}.", jobEntity.getId(),
-                        event.getStatus().convertTaskStatus());
-            }
-        } else {
-            Optional<TaskEntity> taskEntity = taskRepository.findByJobId(jobEntity.getId());
-            if (taskEntity.isPresent() && !taskEntity.get().getStatus().isTerminated()) {
-                int row = taskRepository.updateStatusById(taskEntity.get().getId(),
+                log.info("Update taskTask successfully, taskId={}, status={}.", jobEntity.getId(),
```
Comment (krihy) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/DefaultJobTerminateListener.java:91]: jobEntity.getId() -> taskEntity.getId()

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/JobType.java [hunk 3]
```diff
@@ -32,5 +32,10 @@ public enum JobType {
 
     DATA_ARCHIVE_ROLLBACK,
 
-    ONLINE_SCHEMA_CHANGE_COMPLETE
+    ONLINE_SCHEMA_CHANGE_COMPLETE;
+
+    public boolean executeInTaskFramework() {
```
Comment (krihy) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/JobType.java:37]: executeInTaskFramework -> isJointTaskFramework

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/CloudDLMJobStore.java (hunk 1 above)
Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/CloudDLMJobStore.java:148]: This is not necessary as this configuration is dynamic

**Review by smallsheeeep — DISMISSED** [review]
LGTM

**Review by yizhouxw — APPROVED** [review]
just approve, review by others

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/CloudDLMJobStore.java [hunk 4]
```diff
@@ -15,37 +15,117 @@
...
 import com.oceanbase.tools.migrator.core.meta.TenantMeta;
 
+import lombok.extern.slf4j.Slf4j;
+
 /**
  * @Author：tinker
  * @Date: 2024/1/24 19:57
  * @Descripition:
  */
+@Slf4j
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/CloudDLMJobStore.java:53]: please rename it~

**Review by krihy — APPROVED** [review]
lgtm

**Review by smallsheeeep — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**